numba/core/types/npytypes.py

```diff
@@ -452,9 +452,8 @@ def copy(self, dtype=None, ndim=None, layout=None, readonly=None):
             layout = self.layout
         if readonly is None:
             readonly = not self.mutable
-        cls = type(self)
-        return cls(dtype=dtype, ndim=ndim, layout=layout, readonly=readonly,
-                   aligned=self.aligned)
+        return Array(dtype=dtype, ndim=ndim, layout=layout, readonly=readonly,
+                     aligned=self.aligned)
```

Comment:
Note: This is as 0.53 https://github.com/numba/numba/blob/release0.53/numba/core/types/npytypes.py#L441-L451